examples/08_join_aggregation.py

```diff
@@ -1,302 +1,548 @@
 
 .. |Pipeline| replace::
      :class:`~sklearn.pipeline.Pipeline`
 
-.. |GridSearchCV| replace::
-     :class:`~sklearn.model_selection.GridSearchCV`
+.. |HGBC| replace::
+     :class:`~sklearn.ensemble.HistGradientBoostingClassifier`
+
+.. |TunedThresholdClassifierCV| replace::
+     :class:`~sklearn.model_selection.TunedThresholdClassifierCV`
+
+.. |CalibrationDisplay| replace::
+     :class:`~sklearn.calibration.CalibrationDisplay`
 
-.. |TimeSeriesSplit| replace::
-     :class:`~sklearn.model_selection.TimeSeriesSplit`
+.. |pandas.melt| replace::
+     :func:`~pandas.melt`
 
-.. |HGBR| replace::
-     :class:`~sklearn.ensemble.HistGradientBoostingRegressor`
 """
 
-###############################################################################
+# %%
 # The data
 # --------
 #
-# We begin with loading the ratings table from MovieLens.
-# Note that we use the light version (100k rows).
+# We begin with loading the table from figshare. It has around 100k rows.
+from skrub.datasets import fetch_figshare
+
+X = fetch_figshare("48931237").X
+
+# %%
+# The total price is the sum of the price per unit of each product in the basket,
+# multiplied by their quantity. This will also allow us to define a utility function
+# later, in addition of being a useful feature for the learner.
+import numpy as np
 import pandas as pd
 
-from skrub.datasets import fetch_movielens
+from skrub import TableReport
+
+
+def total_price(X):
+    total_price = pd.Series(np.zeros(X.shape[0]), index=X.index, name="total_price")
+    max_item = 24
+    for idx in range(1, max_item + 1):
+        total_price += X[f"cash_price{idx}"].fillna(0) * X[
+            f"Nbr_of_prod_purchas{idx}"
+        ].fillna(0)
 
-ratings = fetch_movielens(dataset_id="ratings")
-ratings = ratings.X.sort_values("timestamp").reset_index(drop=True)
-ratings["timestamp"] = pd.to_datetime(ratings["timestamp"], unit="s")
+    return total_price
 
-X = ratings[["userId", "movieId", "timestamp"]]
-y = ratings["rating"]
-X.shape, y.shape
-###############################################################################
-X.head()
 
-###############################################################################
-# Encoding the timestamp with a TableVectorizer
-# ---------------------------------------------
+X["total_price"] = total_price(X)
+TableReport(X)
+
+# %%
+# Metrics
+# -------
+#
+# To consider the problem from a business perspective, we define our utility function
+# by the cost matrix in the function ``credit_gain_score``. False positive and false
+# negative predictions incur a negative gain.
 #
-# Our first step is to extract features from the timestamp, using the
-# |TableVectorizer|. Natively, it uses the |DatetimeEncoder| on datetime
-# columns, and doesn't interact with numerical columns.
-from skrub import DatetimeEncoder, TableVectorizer
+# Ultimately, we want to maximize this metric. To do so, we can train our learner to
+# minimize a proper scoring rule like the log loss.
```

Comment:
actually if the goal of the example is to show how to use the aggjoiner I wonder if the business metric / proper scoring rule stuff does not distract from that? it's definitely interesting in its own right but it complicates a bit the example (we need to define more functions, use the metadata routing and what not) and does not yield a better understanding of the joiner AFAICT

Reply:
but it does make for a more realistic example, and we need some of those to check that skrub has enough flexibility for real-world scenarios and not just toy examples :thinking: